python interface set_software(arg1, arg2) breaks readability

**Brief description**
Following the `firstwrite.py`and `firstread.py`introduction, one could also use the `set_software` method to specify the software used and since recently define the software version as second argument. Specifying the second argument however breaks the readability of the openPMD files (at least with ADIOS2).  

**To Reproduce**


```python
import openpmd_api as io
import numpy as np

# write file
series = io.Series("test_simData_%06T.bp", io.Access_Type.create)
i = series.iterations[1000]

series.set_author("Jim Knopf")
series.set_attribute("dinner", "Pizza and Coke")

series.set_software("jupyter", "abc_1.2.3")

series.flush()
del series

# read file
series = io.Series("test_simData_%06T.bp", io.Access_Type.read_only)
```

leads to the following error:
```
---------------------------------------------------------------------------
RuntimeError                              Traceback (most recent call last)
<ipython-input-124-4ca1f3f8499d> in <module>
----> 1 series = io.Series("test_simData_%06T.bp", io.Access_Type.read_only)

RuntimeError: [ADIOS2] Requested attribute (/openPMD) not found in backend.
```

The same happens if the default version is used:
```python
series.set_software("jupyter")
```

**Expected behavior**
However not using `set_software`:
```python
import openpmd_api as io
import numpy as np

# write file
series = io.Series("test_simData_%06T.bp", io.Access_Type.create)
i = series.iterations[1000]

series.set_author("Jim Knopf")
series.set_attribute("dinner", "Pizza and Coke")


series.flush()
del series

# read file
series = io.Series("test_simData_%06T.bp", io.Access_Type.read_only)
i = series.iterations[1000]

print(series.openPMD)
print(series.author)
print(series.software)
print(series.software_version)
print(series.get_attribute("dinner"))
```

returns
```
1.1.0
Jim Knopf
openPMD-api
0.12.0-alpha
Pizza and Coke
```

using the deprecated approach also works:
```python
import openpmd_api as io
import numpy as np

# write file
series = io.Series("test_simData_%06T.bp", io.Access_Type.create)
i = series.iterations[1000]

series.set_author("Jim Knopf")
series.set_attribute("dinner", "Pizza and Coke")

series.set_software("jupyter")
series.set_software_version("3.1415")

series.flush()
del series

# read file
series = io.Series("test_simData_%06T.bp", io.Access_Type.read_only)
i = series.iterations[1000]

print(series.openPMD)
print(series.author)
print(series.software)
print(series.software_version)
print(series.get_attribute("dinner"))
```

returns
```
1.1.0
Jim Knopf
jupyter
3.1415
Pizza and Coke
```


**Software Environment**
 - version of openPMD-api: 0.12.0-alpha
 - installed openPMD-api via: conda-forge
 - operating system: CentOS Linux 7
 - machine: hemera/HZDR
 - name and version of Python implementation: 3.8.6 (packaged by conda-forge)
 - version of HDF5: 1.10.6
 - version of ADIOS1: 1.13.1
 - version of ADIOS2: 2.6.0


**Additional context**
While implementing the test for the missing `machine` attribute in #796, I discovered that the `software`and `software_version` attributes were set bet never checked. Implementing the check resulted in errors.

I am able to fail the unit test even with JSON, so this is more of a internal logic issue per se (or we delay all backends' attribute writes as a work-around):
```diff
diff --git a/test/python/unittest/API/APITest.py b/test/python/unittest/API/APITest.py
index d2428436..17ec8643 100644
--- a/test/python/unittest/API/APITest.py
+++ b/test/python/unittest/API/APITest.py
@@ -117,6 +117,7 @@ class APITest(unittest.TestCase):
         self.assertEqual(len(i.particles), 0)
 
     def attributeRoundTrip(self, file_ending):
+        print("attributeRoundTrip with {0}".format(file_ending))
         # write
         series = io.Series(
             "unittest_py_API." + file_ending,
@@ -124,8 +125,8 @@ class APITest(unittest.TestCase):
         )
 
         # meta data
-        series.set_software("nonsense")  # with unspecified version
-        series.set_software_version("1.2.3")  # deprecated
+        #series.set_software("nonsense")  # with unspecified version
+        #series.set_software_version("1.2.3")  # deprecated
         series.set_software("openPMD-api-python-tests", "0.42.0")
         series.machine = "testMachine"
 
@@ -254,6 +255,8 @@ class APITest(unittest.TestCase):
             io.Access.read_only
         )
 
+        self.assertEqual(series.software, "openPMD-api-python-tests")
+        self.assertEqual(series.software_version, "0.42.0")
         self.assertEqual(series.machine, "testMachine")
 
         self.assertEqual(series.get_attribute("char"), "c")
```
```
======================================================================
FAIL: testAttributes (API.APITest.APITest)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "openPMD-api/test/python/unittest/API/APITest.py", line 376, in testAttributes
    self.attributeRoundTrip(ext)
  File "openPMD-api/test/python/unittest/API/APITest.py", line 258, in attributeRoundTrip
    self.assertEqual(series.software, "openPMD-api-python-tests")
AssertionError: 'openPMD-api' != 'openPMD-api-python-tests'
- openPMD-api
+ openPMD-api-python-tests
```

Update: I checked the created file `unittest_py_API.json` and the attribute is properly written. Must be a read issue then.

--- a/src/Series.cpp
+++ b/src/Series.cpp
@@ -25,6 +25,7 @@
 #include "openPMD/IO/AbstractIOHandlerHelper.hpp"
 #include "openPMD/IO/Format.hpp"
 #include "openPMD/Series.hpp"
+#include "openPMD/version.hpp"
 
 #include <exception>
 #include <iomanip>
@@ -533,24 +534,16 @@ Series::init(std::shared_ptr< AbstractIOHandler > ioHandler,
 void
 Series::initDefaults()
 {
-    std::stringstream openPMDstandard;
-    openPMDstandard << OPENPMD_STANDARD_MAJOR << "."
-                    << OPENPMD_STANDARD_MINOR << "."
-                    << OPENPMD_STANDARD_PATCH;
-    setOpenPMD( openPMDstandard.str() );
-    setOpenPMDextension(0);
-    setAttribute("basePath", std::string(BASEPATH));
-    setDate( auxiliary::getDateString() );
-
-    std::stringstream openPMDapi;
-    openPMDapi << OPENPMDAPI_VERSION_MAJOR << "."
-               << OPENPMDAPI_VERSION_MINOR << "."
-               << OPENPMDAPI_VERSION_PATCH;
-    if( std::string( OPENPMDAPI_VERSION_LABEL ).size() > 0 )
-        openPMDapi << "-" << OPENPMDAPI_VERSION_LABEL;
-    setSoftware( "openPMD-api", openPMDapi.str() );
-
-    // TODO Potentially warn on flush if software and author are not user-provided (defaulted)
+    if( !containsAttribute("openPMD"))
+        setOpenPMD( getStandard() );
+    if( !containsAttribute("openPMDextension"))
+        setOpenPMDextension(0);
+    if( !containsAttribute("basePath"))
+        setAttribute("basePath", std::string(BASEPATH));
+    if( !containsAttribute("date"))
+        setDate( auxiliary::getDateString() );
+    if( !containsAttribute("software"))
+        setSoftware( "openPMD-api", getVersion() );
 }
 
 template< typename IterationsContainer >

--- a/test/python/unittest/API/APITest.py
+++ b/test/python/unittest/API/APITest.py
@@ -254,6 +254,8 @@ class APITest(unittest.TestCase):
             io.Access.read_only
         )
 
+        self.assertEqual(series.software, "openPMD-api-python-tests")
+        self.assertEqual(series.software_version, "0.42.0")
         self.assertEqual(series.machine, "testMachine")
 
         self.assertEqual(series.get_attribute("char"), "c")

